### PR TITLE
Support ActiveModel::Errors as implemented in v. 6.1

### LINKED ISSUE #75: Incompatible with ActiveModel 6.1
The ActiveModel::Errors class has changed, keeping most of the high level API compatible, but some of the underlying implementation are not working anymore.

The class `ActiveData::Model::Validations::NestedValidator` is not compatible with the new implementation, it should be something on the lines of:

```rb
def self.validate_nested(record, name, value)
  if value.is_a?(Enumerable)
    value.each.with_index do |object, i|
      if yield(object)
        object.errors.each do |error|
          record.errors.import(error, key: "#{name}.#{i}.#{error.key}")
        end
      end
    end
  elsif value && yield(value)
    value.errors.each do |error|
      record.errors.import(error, key: "#{name}.#{error.key}")
    end
  end
end
```

but of course this only works with the new errors interface… so maybe a version check? I haven't been able to find a version-independent way to achieve it.

### PR DESCRIPTION
**Pull request by rewritten**
Closes #75

### CODE REVIEW

File: lib/active_data/model/validations/nested.rb [hunk 1]
```diff
@@ -21,6 +23,22 @@ def self.validate_nested(record, name, value)
...
+            value.each.with_index do |object, i|
+              if yield(object)
+                object.errors.each do |error|
+                  record.errors.import(error, attribute: "#{name}.#{i}.#{error.attribute}")
+                end
+              end
+            end
+          elsif value && yield(value)
+            value.errors.each do |error|
+              record.errors.import(error, attribute: "#{name}.#{i}.#{error.attribute}")
```
Comment (rewritten) [inline lib/active_data/model/validations/nested.rb:38]:
```suggestion
              record.errors.import(error, attribute: "#{name}.#{error.attribute}")
```

**Comment by pyromaniac** [comment]
Hey, thanks for the fix! Do I trust you and everything is green in 6.1 and before 6.1? It looks like I need to migrate to GHA to see the real results.

**Comment by rewritten** [comment]
Yep, you should never trust me.

For the record, I used the same trick with a prepended module in a PR to toptal/granite#62, and tested it with 6.0 (old API) and 6.1 (new API) manually.

**Comment by rewritten** [comment]
Closed in favor of #77

**Pull request closed without merging**